napari/_vispy/_tests/test_vispy_calls.py

```diff
@@ -11,7 +11,9 @@ def test_data_change_ndisplay_image(viewer_factory):
     layer = viewer.add_image(data)
     visual = view.layer_to_visual[layer]
 
-    @patch.object(visual, '_on_data_change', wraps=visual._on_data_change)
+    @patch.object(
+        visual, '_on_slice_data_change', wraps=visual._on_slice_data_change
+    )
```

Comment:
From https://docs.python.org/3/library/unittest.mock.html#patch-object here you can "patch the named member (attribute) on an object (target) with a mock object." i.e. patching on `_on_slice_data_change` the `visual` object so that we can do things like count the number of times it gets called. Because we're using `wraps` too, we'll actually make the function call at the same time. This allows us to count the calls and also have the needed code execute